### PR TITLE
always use max cmake when cmake3 and cmake are all existed

### PR DESCRIPTION
**Pull request by XiaobingSuper**
For Pytorch source build when using Ninja generator, it requires **CMake >=3.13**,  Pytorch always checks **cmake3 >= 3.10** first, so when **3.13> cmake3 >= 3.10** and then PyTorch will use cmake3, there will report an error: ```Using the Ninja generator requires CMake version 3.13 or greater```  even the **CMake >=3.13** .  

For example: for my centos machine, the system CMake3 is ```3.12```,  and my conda env's CMake is ```3.19.6```,  there will have a build error which PyTorch choose CMake 3, I can update CMake3 or create an alias or a symlink to solve this problem, but the more reasonable way is that ```_get_cmake_command ``` always return the newest CMake executable (unless explicitly overridden with a same CMAKE_PATH environment variable).

### CODE REVIEW

**Comment by XiaobingSuper** [comment]
@gchanan, could you help review this PR? I think the user don't need to upgrade the system's cmake3 when conda's cmake has meet the build‘s requirements.

**Comment by XiaobingSuper** [comment]
@malfet @ezyang

**Review by malfet — CHANGES_REQUESTED** [review]
Thank you for proposing the change. Overall change looks good to me, but can you please make a few BE changes:
- Instead of repeating `distutils.version.LooseVersion("3.10.0")` turn into a `_CMAKE_MIN_VERSION` constant
- Perhaps modify `CMake._get_version` to accept None and return None as result
Then the code will look much simpler.
Also, please modify PR description to jsut state, that `_get_cmake_command` should return newest cmake executable (unless explicitly overriden with same `CMAKE_PATH` environment variable)

And description about "updating cmake requires root privileges" is not accurate, one can always create an alias or a symlink

File: tools/setup_helpers/cmake.py [hunk 1]
```diff
@@ -118,20 +118,31 @@ def _get_cmake_command() -> str:
...
             return cmake_command
-        cmake3 = which('cmake3')
-        cmake = which('cmake')
-        if cmake3 is not None and CMake._get_version(cmake3) >= LooseVersion("3.10.0"):
+        cmake3_version = CMake._get_version(which('cmake3'))
+        cmake_version = CMake._get_version(which('cmake'))
+
+        _cmake_min_version = LooseVersion("3.10.0")
+        if (cmake3_version is None or cmake3_version < _cmake_min_version) and \
+                (cmake_version is None or cmake_version < _cmake_min_version):
```
Comment (malfet) [inline tools/setup_helpers/cmake.py:126]:
```suggestion
        if all(ver is None or ver < _cmake_min_version) for ver in [cmake_version, cmake3_version]):
```

**Review by malfet — APPROVED** [review]

File: tools/setup_helpers/cmake.py (hunk 1 above)
Comment (XiaobingSuper) [reply tools/setup_helpers/cmake.py:126]: Chnaged.

**Pull request closed without merging**